crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -0,0 +1,696 @@
+    quote_id: i64,
+    domain_separator: DomainSeparator,
+    settlement_contract: H160,
+) -> Result<Option<(OnchainOrderPlacement, Order)>> {
+    let (order_data, owner, signing_scheme, order_uid) =
+        extract_order_data_from_onchain_order_placement_event(order_placement, domain_separator)?;
+
+    let quote_kind = convert_signing_scheme_into_quote_kind(signing_scheme, false)
+        .map_err(|err| anyhow!("Error invalid signature transformation: {:?}", err))?;
+
+    let parameters = QuoteSearchParameters {
+        sell_token: H160::from(order_data.sell_token.0),
+        buy_token: H160::from(order_data.buy_token.0),
+        sell_amount: order_data.sell_amount,
+        buy_amount: order_data.buy_amount,
+        fee_amount: order_data.fee_amount,
+        kind: order_data.kind,
+        // Original quote was made from user account, and not necessarily from owner.
+        from: order_placement.sender,
+        app_data: order_data.app_data,
+        quote_kind,
+    };
+    let quote = get_quote_and_check_fee(
+        quoter,
+        &parameters.clone(),
+        Some(quote_id as i64),
+        order_data.fee_amount,
+        model::quote::QuoteSigningScheme::Eip1271 {
+            onchain_order: true,
+        },
+    )
+    .await
+    .map_err(|err| {
+        anyhow!(
+            "Error while fetching the quote {:?} error: {:?}",
+            parameters,
+            err
+        )
+    })?;
+
+    let full_fee_amount = quote.data.fee_parameters.unsubsidized();
+
+    // Orders that are placed and priced outside the market (i.e. buying
+    // more than the market can pay or selling less than the market wants)
+    // get flagged as liquidity orders. The reasoning is that these orders
+    // are not intended to be filled immediately and so need to be treated
+    // slightly differently by the protocol.
+    let is_liquidity_order =
+        if is_order_outside_market_price(&parameters.sell_amount, &parameters.buy_amount, &quote) {
+            tracing::debug!(%order_uid, ?owner, "order being flagged as outside market price");
+            true
+        } else {
+            false
+        };
+
+    let order = database::orders::Order {
+        uid: ByteArray(order_uid.0),
+        owner: ByteArray(owner.0),
+        creation_timestamp: chrono::offset::Utc::now(),
+        sell_token: ByteArray(order_data.sell_token.0),
+        buy_token: ByteArray(order_data.buy_token.0),
+        receiver: order_data.receiver.map(|h160| ByteArray(h160.0)),
+        sell_amount: u256_to_big_decimal(&order_data.sell_amount),
+        buy_amount: u256_to_big_decimal(&order_data.buy_amount),
+        valid_to: order_data.valid_to as i64,
+        app_data: ByteArray(order_data.app_data.0),
+        fee_amount: u256_to_big_decimal(&order_data.fee_amount),
+        kind: order_kind_into(order_data.kind),
+        partially_fillable: order_data.partially_fillable,
+        signature: order_placement.signature.1 .0.clone(),
```

Comment:
Nit:

```suggestion
        signature: order_placement.signature.1.0.clone(),
```

I have to say I'm not sure what `.1.0` is expected to be, even if I can guess.

Reply:
It's the signature data. You are right that it is quite hard to read, but its due to the solidity->contract->data conversions that are automated. I think there is nothing I can do about it